src/Handler/CurlFactory.php

```diff
@@ -227,11 +227,9 @@ private function applyMethod(EasyHandle $easy, array &$conf)
         }
 
         $method = $easy->request->getMethod();
-        if ($method === 'PUT' || $method === 'POST') {
+        if (($method === 'PUT' || $method === 'POST') && !$easy->request->hasHeader('Content-Length')) {
```

Comment:
Sorry, I meant this thing too. Right now if the method is PUT or POST, but the second condition evaluates to false the elseif condition is still evaluated.